include/envoy/upstream/cluster_manager.h

```diff
@@ -29,6 +30,11 @@
 #include "envoy/upstream/upstream.h"
 
 namespace Envoy {
+
+namespace Config {
+class SubscriptionFactory;
```

Comment:
choose from forward declaration or #include "envoy/config/subscription_factory.h"

Reply:
I thought there was a circular dependency, but apparently not.